src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -54,17 +54,15 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
                     chain.ChainPolicy.ExtraStore.AddRange(additionalCertificates);
                 }
 
-                if (trust != null)
+                if (trust?._store?.Certificates.Count > 0)
                 {
                     chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
-                    if (trust._store != null)
-                    {
-                        chain.ChainPolicy.CustomTrustStore.AddRange(trust._store.Certificates);
-                    }
-                    if (trust._trustList != null)
-                    {
-                        chain.ChainPolicy.CustomTrustStore.AddRange(trust._trustList);
-                    }
```

Comment:
I recommend returning to the previous pattern of just calling AddRange.

Then, you can decide to only change the custom trust mode if chain.ChainPolicy.CustomTrustStore is not empty:


```C#
if (chain.ChainPolicy.CustomTrustStore.Count > 0)
{
    chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
}
```

Reply:
It's not intuitive to me, though, that we'd let "I said empty custom trust" mean "so use system trust" on Android, but mean "so nothing is trusted" on other platforms.

The trust being specified as empty should really manifest as an error here, or somewhere else, not mean "just do something different".